profile_bluesky/startup/instrument/devices/ad_imm_plugins.py

```diff
@@ -51,37 +51,39 @@ class IMMoutLocal(Device):
     enable = Component(
         EpicsSignalWithRBV, "EnableCallbacks", string=True, kind="config"
     )
-    file_format = Component(
-        EpicsSignalWithRBV, "NDFileIMM_format", string=True, kind="config"
-    )
+    # FIXME: replaced 8LAMBDA1:IMM1:NDFileIMM_format with _________ something lua something (ask Keenan about this) 8LAMBDA1:set1:userScript1.A
+    # 8LAMBDA1:set1:userScript1 is a luascript record
+    # lua_script = Component(EpicsSignal, "set1:userScript.A", kind="config")  # but does not belong in the IMM1 plugin
+    # below is the old file_format to delete
+    # file_format = Component(
+    #     EpicsSignalWithRBV, "NDFileIMM_format", string=True, kind="config"
+    # )
     file_name = Component(EpicsSignalWithRBV, "FileName", string=True, kind="config")
     file_number = Component(EpicsSignalWithRBV, "FileNumber", kind="config")
     file_path = Component(EpicsSignalWithRBV, "FilePath", string=True, kind="config")
-    # full_file_name = Component(
-    #     EpicsSignalRO, "FullFileName_RBV", string=True, kind="config"
-    # )
+    full_file_name = Component(
+        EpicsSignalRO, "FullFileName_RBV", string=True, kind="config"
+    )
     num_capture = Component(EpicsSignalWithRBV, "NumCapture", kind="config")
     num_captured = Component(EpicsSignalRO, "NumCaptured_RBV")
-    num_pixels = Component(EpicsSignalRO, "NDFileIMM_num_imm_pixels_RBV", kind="config")
+    # num_pixels = Component(EpicsSignalRO, "NDFileIMM_num_imm_pixels_RBV", kind="config")
 
-    unique_id = Component(EpicsSignalRO, "NDFileIMM_uniqueID_RBV")
+    # unique_id = Component(EpicsSignalRO, "NDFileIMM_uniqueID_RBV")
 
 
 class IMM_DeviceMixinBase(Device):
     """
     attributes used by area detectors using the IMM Plugin
     """
-
-    immout = Component(IMMoutLocal, "IMMout:")
-    imm1 = Component(IMMnLocal, "IMM1:")
-    imm2 = Component(IMMnLocal, "IMM2:")
+    # imm1 = Component(IMMnLocal, "IMM1:")
+    imm1 = Component(IMMoutLocal, "IMM1:")
 
     def setIMM_Cmprs(self):
         """
         Set all IMM plugins for compression.
         """
         # from SPEC macro: ccdset_compr_params_ad_Lambda
-        for plugin in (self.imm1, self.imm2, self.immout):
+        for plugin in (self.imm1):
             if plugin.file_format.get() not in (1, "IMM_Cmprs"):
```

Comment:
there is no `file_format` attribute now - what to do here?

Reply:
To self: Check with Suresh whether `setIMM_Cmprs` and `setIMM_Raw` are actually being used

Reply:
In the code, only appears here (so, not in use anywhere else in bluesky).

Reply:
comment out now, consider to delete later (after this PR is merged)